pincer/objects/guild/guild.py

```diff
@@ -1916,6 +1919,264 @@ async def get_webhooks(self) -> AsyncGenerator[Webhook, None]:
+        scheduled_start_time : Optional[:class:`datetime`]
+            The scheduled start time of the event.
+        scheduled_end_time : Optional[:class:`datetime`]
+            The scheduled end time of the event.
+        entity_metadata : Optional[:class:`str`]
+            The metadata of the entity to be scheduled.
+        channel_id : Optional[:class:`int`]
+            The channel id of the channel to be scheduled.
+        description : Optional[:class:`str`]
+            The description of the scheduled event.
+        status : Optional[:class:`int`]
+            The status of the scheduled event.
+        reason : Optional[:class:`str`]
+            The reason for modifying the scheduled event.
+
+        Raises
+        ------
+        :class:`ValueError`
+            If the scheduled event is in the past,
+            or if the scheduled end time is before the scheduled start time.
+
+        Returns
+        -------
+        :class:`~pincer.objects.guild.scheduled_event.ScheduledEvent`
+            The scheduled event object.
+        """
+        if scheduled_start_time:
+            if scheduled_start_time < datetime.now():
+                raise ValueError("An event cannot be created in the past")
+
+            if (
+                scheduled_end_time
+                and scheduled_end_time < scheduled_start_time
+            ):
+                raise ValueError("An event cannot start before it ends")
+
+        kwargs: Dict[str, str] = remove_none(
+            {
+                "name": name,
+                "privacy_level": privacy_level,
+                "entity_type": entity_type,
+                "scheduled_start_time": scheduled_start_time.isoformat()
+                if scheduled_start_time is not None
+                else None,
+                "scheduled_end_time": scheduled_end_time.isoformat()
+                if scheduled_end_time is not None
+                else None,
+                "entity_metadata": entity_metadata,
+                "channel_id": channel_id,
+                "description": description,
+                "status": status,
+            }
+        )
+
+        data = await self._http.patch(
+            f"guilds/{self.id}/scheduled-events/{_id}",
+            data=kwargs,
+            headers={"X-Audit-Log-Reason": reason},
+        )
+        return ScheduledEvent.from_dict(data)
+
+    async def delete_scheduled_event(self, _id: int):
+        """
+        Delete a scheduled event.
+
+        Parameters
+        ----------
+        _id : :class:`int`
+            The id of the scheduled event.
+        """
+        await self._http.delete(f"guilds/{self.id}/scheduled-events/{_id}")
+
+    async def get_guild_scheduled_event_users(
+        self,
+        _id: int,
+        limit: int = 100,
+        with_member: bool = False,
+        before: Optional[int] = None,
+        after: Optional[int] = None,
+    ) -> AsyncGenerator[GuildScheduledEventUser, None]:
```

Comment:
Same as above I guess?
```suggestion
    ) -> AsyncIterator[GuildScheduledEventUser]:
```